tables/tests/test_indexvalues.py

```diff
@@ -2316,13 +2316,18 @@ def test12a(self):
         self.assertEqual(results1, results2)
 
         # Second selection: bool
+        # TODO: Fails for FINSV2aTestCase
```

Comment:
@maxnoe @avalentino this PR is very close to working. This is the one test I couldn't figure out how to fix. There is something wrong with the `results1 = [p["var2"] for p in table1.where('t1var2 == True')]` -- there are four entries that are `True` but it only finds two of them :( Any help here would be greatly appreciated.

It can be replicated locally "just" by building against NumPy 2.0 (e.g., on this branch and with NumPy 2.0 on your local system) and running the test.

Reply:
Is that maybe an issue of python bool vs. numpy's np.bool_ ?

Reply:
I thought it could be... tried various tricks here but couldn't get it to work. Feel free to try some stuff locally.

Don't know if this is informative:
```
>>> np.bool(True) == True == np.bool_(True)
np.True_
```
maybe it's because this evaluates to `np.True_` instead of `True`?